tests/integration/security/rbac/v2/basic/basic_test.go

```diff
@@ -90,6 +90,7 @@ func TestRBACV2(t *testing.T) {
 				Galley:         g,
 				Pilot:          p,
 			})
+			a.WaitUntilReadyOrFail(t, b, c, d)
```

Comment:
Should this be rolled into echoboot.NewOrFail? I am worried that new-then-wait paradigm is prone to introduce bugs.

You can have an NewOrFail overload that creates multiple echo instances at the same time, to keep the contention/wait.

Reply:
No, it can't be unfortunately. The problem is that the wait times for deployment would get stacked up.  In order to allow the deployment to be parallel, we have to wait after the fact.

Reply:
That was my point with the overload comment. You can make echoboot.NewOrFail to take multiple echo.Configs and multiple results. That allows you to encapsulate the wait in the NewOrFail code.

Another alternative is to change the return type. You can return the spiritual equivalent of Future<Echo>, where the caller will need to wait and extract. Or a similar solution would be to return Echo, and a channel that the caller can wait for its close.

My worry is, most of our API is based on dependability on call-return being a meaningful sync point. When Pilot or Galley NewOrFail returns, you *know* that you can now use it. Echo breaks the convention, which makes is likelier to introduce bugs because of this.

Reply:
I agree, I just don't think it's something that should be tackled in this PR.  The right API will take a bit more thought than I want to invest here.